go/app/main.go

```diff
@@ -26,16 +42,206 @@ func root(c echo.Context) error {
 }
 
 func addItem(c echo.Context) error {
+	var items Items
 	// Get form data
 	name := c.FormValue("name")
-	c.Logger().Infof("Receive item: %s", name)
+	category := c.FormValue("category")
+	image, err := c.FormFile("image")
+	if err != nil {
+		return err
+	}
+
+	src, err := image.Open()
+	if err != nil {
+		return err
+	}
+	defer src.Close()
+
+	// Create a new SHA256 hash
+	hash := sha256.New()
+
+	hashInBytes := hash.Sum(nil)
+
+	// Convert hash bytes to hex string
+	hashString := hex.EncodeToString(hashInBytes)
+
+	image_jpg := hashString + ".jpg"
+
+	new_image, err := os.Create("images/" + image_jpg)
+	if err != nil {
+		return err
+	}
+
+	// Copy the file content to the hash
+	if _, err := io.Copy(new_image, src); err != nil {
+		return err
+	}
+
+	item := Item{Name: name, Category: category, Image: image_jpg}
+
+	c.Logger().Infof("Receive item: %s, %s", item.Name, item.Category, item.Image)
+	message := fmt.Sprintf("item received: %s, %s, %s", item.Name, item.Category, item.Image)
 
-	message := fmt.Sprintf("item received: %s", name)
 	res := Response{Message: message}
 
+	items.Items = append(items.Items, item)
+
+	db, err := sql.Open("sqlite3", "/Users/kitaharuka/mercari-build-training/db/mercari.sqlite3")
```

Comment:
[nits] DBの名前を上部にconstとして定義してもいいかもしれないです

```
const (
	ImgDir = "images"
        DB_PATH = "/Users/kitaharuka/mercari-build-training/db/mercari.sqlite3"
)
```

```suggestion
	db, err := sql.Open("sqlite3", DB_PATH)
```